markdown/markdown.php

```diff
@@ -49,5 +49,25 @@ function markdown_post_local_start(App $a, &$request) {
 		return;
 	}
 
+	// Elements that shouldn't be parsed
+	$elements = ['code', 'noparse', 'nobb', 'pre'];
+	foreach ($elements as $element) {
+		$request['body'] = preg_replace_callback("/\[" . $element . "\](.*?)\[\/" . $element . "\]/ism",
+			function ($match)  use ($element) {
+				return '[base64' . $element . ']' . base64_encode($match[1]) . '[/base64' . $element . ']';
+			},
+			$request['body']
+		);
+	}
+
 	$request['body'] = Markdown::toBBCode($request['body']);
+
+	foreach (array_reverse($elements) as $element) {
```

Comment:
What is the point of this `array_reverse`?

Reply:
You could have nested elements. Then they have to be decoded in the opposite order.